sunpy/io/_fits.py

```diff
@@ -79,11 +79,12 @@ def read(filepath, hdus=None, memmap=None, **kwargs):
         for h in hdulist:
             h.verify('silentfix+warn')
 
-        headers = get_header(hdulist)
         pairs = []
 
-        for i, (hdu, header) in enumerate(zip(hdulist, headers)):
+        for i, hdu in enumerate(hdulist):
             try:
+                hdu.data  # access data to force any scaling BSCALE/BZERO to be applied and keyword will be updated
```

Comment:
I worry about the performance impact and general side-effects of doing this here. :see_no_evil:

Reply:
Yea I was too but I don't _think_ it's any different to this line which was being called `HDPair(hdu.data, header)` it has the same side effect as far as I can see, `hdu.data` is being accessed and causes astropy.io to load and scale the data, I'll try to confirm either way. Long term if you want to support lazy loading and dask etc this is an issue for sure.